trl/trainer/sft_trainer.py

```diff
@@ -169,6 +170,11 @@ def __init__(
                 "You passed a `DataCollatorForCompletionOnlyLM` to the SFTTrainer. This is not compatible with the `packing` argument."
             )
 
+        if not packing and formatting_func is None and dataset_text_field is None and data_collator is not None and not isinstance(data_collator, DataCollatorForSeq2Seq):
+            raise ValueError(
+                "If no formatting_func / dataset_text_field provided, the data_collator should be a `DataCollatorForSeq2Seq` object"
```

Comment:
do we need to mandate this? example users can pass default collator or a custom collator ? I think setting it by default to seq2seq if not set is a good change, but wondering if we need to mandate that it can be only seq2seq (I don t think this code generally handles if users pass the wrong collator - thats their issue? )

Reply:
I'm not sure, I guess one question I have - will this work correctly if provided a `DataCollatorForCompletionOnlyLM`? I think that is the most likely incorrect choice

Reply:
it probably wont, but I think we have to allow custom collators in this code and if user chooses wrong collator its on them - all we can do is set default like you did? I dont think they throw error otherwise currently

like if user today passes seq2seq collator on other alpaca style format, the code here doesnt check for it and throw error - I think by design they left it as user's responsibility to pass correct one and they only provide defaults

Reply:
`DataCollatorForCompletionOnlyLM` wont even fit here API wise, because users have to pass a datatext text field and a response template as arguments for this. So I think with documentation users will know the completionOnly is only meant for datasets with a preformatted datatext field. 

so if they try to use it, the error should come come from collator that required fields are not passed. 

but if they pass `DataCollatorForLanguageModeling` , I dont know what the bahvior will be :D need to confirm

 maybe we can check `         if  dataset_text_field is None and data_collator is not None and isinstance(data_collator,DataCollatorForLanguageModeling ): return error or something
 
 maybe we can add a TODO here to think about thsi check further and merge for now?

Reply:
They do a little bit of validation here to ensure that you can't use `DataCollatorForCompletionOnlyLM` with `packing` - that is why I added this here specifically

```
        if packing and data_collator is not None and isinstance(data_collator, DataCollatorForCompletionOnlyLM):
            raise ValueError(
                "You passed a `DataCollatorForCompletionOnlyLM` to the SFTTrainer. This is not compatible with the `packing` argument."
            )
```

but sounds good, for now I can remove this check / test and add a comment instead